components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -0,0 +1,282 @@
+  _selected = selected;
+  if (selected) {
+    self.label.textColor = self.selectedItemTintColor;
+    self.iconImageView.image = self.selectedImage;
+
+    switch (self.titleHideState) {
+      case MDCBottomNavigationBarTitleHideStateDefault:
+        self.label.hidden = NO;
+        break;
+      case MDCBottomNavigationBarTitleHideStateNever:
+        self.label.hidden = NO;
+        break;
+      case MDCBottomNavigationBarTitleHideStateAlways:
+        self.label.hidden = YES;
+        break;
+    }
+  } else {
+    self.label.textColor = self.unselectedItemTintColor;
+    self.iconImageView.image = self.unselectedImage;
+
+    switch (self.titleHideState) {
+      case MDCBottomNavigationBarTitleHideStateDefault:
+        self.label.hidden = YES;
+        break;
+      case MDCBottomNavigationBarTitleHideStateNever:
+        self.label.hidden = NO;
+        break;
+      case MDCBottomNavigationBarTitleHideStateAlways:
+        self.label.hidden = YES;
+        break;
+    }
+  }
+  [self centerLayoutAnimated:animated];
+}
+
+- (void)setCircleHighlightHidden:(BOOL)circleHighlightHidden {
+  if (!circleHighlightHidden) {
+    self.circleLayer.opacity = kMDCBottomNavigationItemViewCircleOpacity;
+    self.iconImageView.image = self.selectedImage;
+  } else {
+    self.circleLayer.opacity = 0;
+    if (self.selected) {
+      self.iconImageView.image = self.selectedImage;
+    } else {
+      self.iconImageView.image = self.unselectedImage;
+    }
+  }
+}
+
+- (void)setSelectedItemTintColor:(UIColor *)selectedItemTintColor {
+  _selectedItemTintColor = selectedItemTintColor;
+  self.label.textColor = self.selectedItemTintColor;
+  self.selectedImage = [self colorizeImage:self.image color:self.selectedItemTintColor];
+  self.circleLayer.fillColor = self.selectedItemTintColor.CGColor;
+}
+
+- (void)setUnselectedItemTintColor:(UIColor *)unselectedItemTintColor {
+  _unselectedItemTintColor = unselectedItemTintColor;
+  self.unselectedImage = [self colorizeImage:self.image color:self.unselectedItemTintColor];
+}
+
+- (void)setBadgeColor:(UIColor *)badgeColor {
+  _badgeColor = badgeColor;
+  self.badge.badgeColor = badgeColor;
+}
+
+- (void)setBadgeValue:(NSString *)badgeValue {
+  _badgeValue = badgeValue;
+  self.badge.badgeValue = badgeValue;
+  if (_badgeValue == nil || _badgeValue.length == 0) {
+    self.badge.hidden = YES;
+  } else {
+    self.badge.hidden = NO;
+  }
+}
+
+- (void)setImage:(UIImage *)image {
+  _image = image;
+
+  self.selectedImage = [self colorizeImage:image color:self.selectedItemTintColor];
+  self.unselectedImage = [self colorizeImage:image color:self.unselectedItemTintColor];
+
+  self.iconImageView.image = self.unselectedImage;
+  [self.iconImageView sizeToFit];
+}
+
+- (void)setTitle:(NSString *)title {
+  _title = [title copy];
+  self.label.text = _title;
+  [self.label sizeToFit];
+}
+
+- (void)setItemTitleFont:(UIFont *)itemTitleFont {
+  _itemTitleFont = itemTitleFont;
+  self.label.font = itemTitleFont;
```

Comment:
We may or may not get a setNeedsLayout call for setting self.label.font.
If we do not get one implicitly , we likely want setNeedsLayout over sizeToFit.

Reply:
Done.